### PR TITLE
Edit README.md and move content to docs/

### PR DESCRIPTION
**Pull request by llzmb**
Changes:
- Move advanced content to docs/.
- Fix links.
- Add links.
- Restructure content.

### CODE REVIEW

**Comment by vanhauser-thc** [comment]
@all: much easier to read at https://github.com/llzmb/AFLplusplus/tree/docs_edit_readme_move_content_to_docs :)

File: README.md [hunk 1]
```diff
@@ -1,1441 +1,181 @@
...
+* If you want to use AFL++ for your academic work, check the [papers page](https://aflplus.plus/papers/) on the website.
+* To cite our work, look at the [Cite](#cite) section.
+* For comparisons, use the fuzzbench `aflplusplus` setup, or use `afl-clang-fast` with `AFL_LLVM_CMPLOG=1`. You can find the `aflplusplus` default configuration on Google's [fuzzbench](https://github.com/google/fuzzbench/tree/master/fuzzers/aflplusplus).
 
 ## Building and installing AFL++
 
-An easy way to install AFL++ with everything compiled is available via docker:
-You can use the [Dockerfile](Dockerfile) (which has gcc-10 and clang-11 -
-hence afl-clang-lto is available!) or just pull directly from the docker hub:
+To install AFL++ with everything compiled, pull the image directly from the Docker Hub:
```
Comment (vanhauser-thc) [inline README.md]: As this doe not really install AFL++, I would rather say "To have AFL++ easily available with everything compiled, ..."

File: README.md [hunk 2]
```diff
@@ -1,1441 +1,181 @@
...
-AFL++ binaries by passing the STATIC=1 argument to make:
-
-```shell
-make STATIC=1
-```
+This image is automatically generated when a push to the stable repo happens (see [docs/branches.md](docs/branches.md)).
+You will find your target source code in `/src` in the container.
 
-These build options exist:
+To build AFL++ yourself, continue at [docs/building_installing.md](docs/building_installing.md).
```
Comment (vanhauser-thc) [inline README.md]: the standard is to call this INSTALL.md (all upper case, same as README.md)

File: README.md [hunk 3]
```diff
@@ -1,1441 +1,181 @@
...
-
-```
-#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
-  // say that the checksum or HMAC was fine - or whatever is required
-  // to eliminate the need for the fuzzer to guess the right checksum
-  return 0;
-#endif
-```
+This is a quick start for fuzzing targets with the source code available.
+To read about the process in detail, see [docs/fuzzing.md](docs/fuzzing.md).
```
Comment (vanhauser-thc) [inline README.md]: we should name this differently. dunno how though.
in_depth.md? expert.md? also the best_practices.md should point to this document directly under the "# best practices" header.

Comment (vanhauser-thc) [reply README.md]: ideas? @domenukk @andreafioraldi @hexcoder-

File: docs/guided_fuzzing.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Challenges of guided fuzzing
```
Comment (vanhauser-thc) [inline docs/guided_fuzzing.md]: remove this file + content please. it is out of context, outdated, and I think does not help anyone.

**Comment by vanhauser-thc** [comment]
some more comments:
the tutorial.md should be linked in the main readme.md

tools.md should be renamed to  -> add_on_tools.md or external_tools or something like that.

the contents of these should all be incorporated into fuzzing.md - having them as single files serves no useful purpose IMHO
choosing_testcases.md 
known_limitations.md
triaging_crashes.md
beyond_crashes.md
in the next call we can discuss where to insert them.

interpreting_output.md and status_screen.md should maybe be combined into one file, but not sure how that should be named.

I am torn about the branches.md file. it is so small I dont think it should be a file on its own. I think just putting that into the main README is easier?

**Comment by vanhauser-thc** [comment]
and the fuzz_approach could be moved to the combined interpreting_output.md and status_screen.md document as well? I think it would fit in there.

File: README.md (hunk 2 above)
Comment (llzmb) [reply README.md]: Thanks for the pointer!

There already is an INSTALL.md: https://github.com/AFLplusplus/AFLplusplus/blob/stable/docs/INSTALL.md
The MacOS X part seems to be the most recent.

I am not sure where to add the content from building_installing.md. For Linux on x86, the INSTALL.md says "there are no special dependencies", but the building_installing.md seems to list them in detail.

File: README.md (hunk 3 above)
Comment (llzmb) [reply README.md]: For now, I went with "fuzzing_expert.md". But this is not set in stone!

**Comment by llzmb** [comment]
@vanhauser-thc, thank you so much for your ideas! I'm going to address the open ones (marked with "rocket") in another branch/pull request.

File: README.md (hunk 2 above)
Comment (vanhauser-thc) [reply README.md]: the original INSTALL.md is mostly outdated. I would say put the MacOS part in a chapter in building_installing.md and then delete INSTALL.md and rename  building_installing.md to INSTALL.md :)

**Comment by vanhauser-thc** [comment]
one more thing - maybe rename behaviour_changes.md to something else? I would not know what is in the file.
maybe "important_changes.md" or "version_changes.md" ... ?

File: README.md (hunk 2 above)
Comment (llzmb) [reply README.md]: Done!

There is a reference to the deleted "section 5" in the source code:
https://github.com/AFLplusplus/AFLplusplus/blob/773baf9391ff5f1793deb7968366819e7fa07adc/GNUmakefile#L544

Can I leave this to you?

Comment (vanhauser-thc) [reply README.md]: yes I will fix that

**Comment by llzmb** [comment]
Just to make sure that everything regarding the "README.md" is done here - this is still on my to-do list for another branch/pull request:
- Add link to "fuzzing_expert.md" in "best_practices.md" (directly under the "# best practices" header).
- Rename "tools.md" to "add_on_tools.md" or "external_tools" or something else that fits better.
- Merge the content of the following files into "fuzzing.md":
  - choosing_testcases.md
  - known_limitations.md
  - triaging_crashes.md
  - beyond_crashes.md
- Merge the content of the following files (name of new file?):
  - interpreting_output.md
  - status_screen.md
  - afl_fuzz_approach.md
- Merge "branches.md" into "README.md" or somewhere else that fits better.

If that's okay with you, can you please merge the pull request?

**Pull request merged**